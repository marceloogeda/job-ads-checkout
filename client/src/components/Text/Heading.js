// @flow

import React from 'react';
import styled from 'styled-components';

// Styles
const Heading = styled.h1`
  font-size: 16px;
`;

export type HeadingProps = {
  children: Node | string
};

export default ({ children }: HeadingProps) => <Heading>{children}</Heading>;
