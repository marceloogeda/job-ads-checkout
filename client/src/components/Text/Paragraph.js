// @flow
import React from 'react';
import styled from 'styled-components';

// Styles
const Paragraph = styled.p`
  font-size: 16px;
`;

export type ParagraphProps = {
  children: Node | string
};

export default ({ children }: ParagraphProps) => (
  <Paragraph>{children}</Paragraph>
);
