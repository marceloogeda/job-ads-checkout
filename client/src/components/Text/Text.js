// @flow
import React, { Component } from 'react';
import styled from 'styled-components';

// Components
import Heading from './Heading';
import Paragraph from './Paragraph';

// Types
import type { Node } from 'react';

// Styles
const TextElement = styled.span`
  font-size: ${props => (!!props.large ? '1.45rem' : '1rem')};
  text-decoration: ${props => (!!props.lineThrough ? 'line-through' : 'none')};
`;

export type TextProps = {
  children: Node,
  large?: boolean,
  lineThrough?: boolean
};

export default class Text extends Component<TextProps> {
  static Heading = Heading;
  static Paragraph = Paragraph;

  render() {
    const { children, large, lineThrough } = this.props;

    return (
      <TextElement large={large} lineThrough={lineThrough}>
        {children}
      </TextElement>
    );
  }
}
