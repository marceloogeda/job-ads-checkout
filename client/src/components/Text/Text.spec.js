import React from 'react';
import TestRenderer from 'react-test-renderer';

import Text from './Text';

describe('# Text', () => {
  it('should render a HTMLSpanElement correctly', () => {
    const testRenderer = TestRenderer.create(<Text>Seek - Job Ads</Text>);

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('should render a HTMLParagraphElement correctly', () => {
    const testRenderer = TestRenderer.create(
      <Text.Paragraph>Seek - Job Ads</Text.Paragraph>
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('should render a HTMLHeadingElement correctly', () => {
    const testRenderer = TestRenderer.create(
      <Text.Heading>Seek - Job Ads</Text.Heading>
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
