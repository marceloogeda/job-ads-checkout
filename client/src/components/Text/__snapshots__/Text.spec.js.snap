// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Text should render a HTMLHeadingElement correctly 1`] = `
<h1
  className="sc-bdVaJa hQJWqK"
>
  Seek - Job Ads
</h1>
`;

exports[`# Text should render a HTMLParagraphElement correctly 1`] = `
<p
  className="sc-bwzfXH iiFAtu"
>
  Seek - Job Ads
</p>
`;

exports[`# Text should render a HTMLSpanElement correctly 1`] = `
<span
  className="sc-htpNat dUUasM"
>
  Seek - Job Ads
</span>
`;
